CONTRIBUTING.md

```diff
@@ -0,0 +1,83 @@
+# Contributing
+
+Before contributing to this repository, discuss the changes through an issue,
+email, or any mode of communciation.
+
+Follow the code of conduct follow in all interactions with the project.
+
+### Reporting Bugs
+
+Bugs are tracked as [GitHub issues](https://guides.github.com/features/issues/).
+
+1. Create an issue and provide the required information in the [issue template](ISSUE_TEMPLATE.md).
+2. Describe the problem with the following details:
+  - A self-explanatory title.
+  - Steps to reproduce the problem. Provide the steps with details such as, what is the step and how is it done.
+  - Examples to demonstrate the steps. Include links to files or reusable snippets that are used in the examples. For providing snippets in the issue, use [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+  - Describe the result you get when you follow the steps, and state the issues, if any.
+  - Explain the expected result and the reason.
+  - Add screenshots of the results of each step.
+3. Add the following configuration and environment details:
+  - Versions of Node and NPM that you use. Run the  `node --version` and `npm --version` command in the terminal to get the correct version.
+  - Name version of the operating system that you use.
```

Comment:
Should there be a comma between name and version? I am assuming that we need to know both sets of information.